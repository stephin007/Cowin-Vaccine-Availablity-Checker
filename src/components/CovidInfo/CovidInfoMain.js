import { useState } from 'react';
import { CssBaseline, Container, Tabs, Tab, AppBar } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

import CovidWorld from './CovidWorld.js/CovidWorld';
import CovidIndia from './CovidIndia.js/CovidIndia';

const useStyles = makeStyles((theme) => ({
	appbar: {
		background: '#16222f',
	},
	section_title: {
		textAlign: 'center',
		textTransform: 'uppercase',
		fontFamily: 'Nunito',
		marginTop: '10px',
	},
}));

const CovidInfo = () => {
	const classes = useStyles();
	const [tabValue, setTabValue] = useState(0);

	const handleTabChange = (value) => {
		setTabValue(value);
	};

	return (
		<>
			<CssBaseline />
			<Container maxWidth="md">
				<h1 className={classes.section_title} data-testId="covidinfo-header">
					World Covid 19 Information
				</h1>
				<AppBar position="static" className={classes.appbar}>
					<Tabs
						value={tabValue}
						variant="fullWidth"
						onChange={(e, val) => handleTabChange(val)}
						data-testId="covidinfo-tabValue"
					>
						<Tab label="World" />
						<Tab label="India" />
					</Tabs>
				</AppBar>
				<CovidIndia value={tabValue} index={1} />
				<CovidWorld value={tabValue} index={0} />
			</Container>
		</>
	);
};

export default CovidInfo;
