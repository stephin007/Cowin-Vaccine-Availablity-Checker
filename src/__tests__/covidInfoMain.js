import { shallow } from 'enzyme';
import React from 'react';
import CovidInfo from '../components/CovidInfo/CovidInfoMain';

describe('render covidInfoMain', () => {
	const wrapper = shallow(<CovidInfo />);
	console.log(wrapper.debug());
	it('render covidinfo correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('renders section title', () => {
		expect(wrapper.find('[data-testId="covidinfo-header"]').text()).toEqual('World Covid 19 Information');
	});
});

describe('check tab value change functionality', () => {
	const wrapper = shallow(<CovidInfo />);
	const event = {
		target: { value: 'success' },
	};
	beforeEach(() => {
		wrapper.find('[data-testId="covidinfo-tabValue"]').simulate('change', event);
	});
	it('change tab value', () => {
		expect(wrapper.find('[data-testId="covidinfo-tabValue"]')).toBeCalledWith('success');
	});
});
