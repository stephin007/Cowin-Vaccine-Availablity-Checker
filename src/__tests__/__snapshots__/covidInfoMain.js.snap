// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render covidInfoMain render covidinfo correctly 1`] = `ShallowWrapper {}`;
